### PR TITLE
Add accessible search field to emoji picker

### PR DESCRIPTION
**Pull request by Pytal**
Before | After
--- | ---
image | image

### CODE REVIEW

**Review by szaimen — CHANGES_REQUESTED** [review]
clicking the x on the right doesnt seem to reset the search results.
image

**Comment by Pytal** [comment]
> What I find a bit confusing is that the input and the first emoji seems to be selected at the same time which is obviously not the case. Otherwise works fine now :)

Seems like this is done in `emoji-mart-vue-fast` itself and the idea was to allow navigating the emojis while still focused onthe input? But this only works when using the built-in search input and not when overriding it

**Comment by raimund-schluessler** [comment]
> > What I find a bit confusing is that the input and the first emoji seems to be selected at the same time which is obviously not the case. Otherwise works fine now :)
> 
> Seems like this is done in `emoji-mart-vue-fast` itself and the idea was to allow navigating the emojis while still focused onthe input? But this only works when using the built-in search input and not when overriding it

This behaviour is anyway a bit weird. While searching, `keydown.right` moves the cursor in the input field *and* the emoji selected at the same time. But `keydown.left` *first* moves the emoji selected and *then* the cursor in the input field. Both that it does two things and then inconsistent is unexpected to me.

However, now `keydown.left/right` does nothing anymore, even after one tabbed into the emoji list. Would be nice if this could be changed.

**Comment by raimund-schluessler** [comment]
> However, now `keydown.left/right` does nothing anymore, even after one tabbed into the emoji list. Would be nice if this could be changed.

Might be a limitation of `emoji-mart-vue-fast`. Their example with a custom search behaves the same.

**Review by raimund-schluessler — APPROVED** [review]

**Review by JuliaKirschenheuter — COMMENTED** [review]
As far i can see Emoji Mart Vue has some a11y problems (incl weird arrow navigation behavior) and to fix them is this is beyond our power ;( for now there is too less time to fix such small part (only status message).
Probably a quick / dirty solution would be to take off this part and just leave a usual input field. What do you mean?

**Comment by Pytal** [comment]
> Probably a quick / dirty solution would be to take off this part and just leave a usual input field. What do you mean?

You mean drop these changes and keep the built-in search input?

**Review by skjnldsv — APPROVED** [review]

File: src/components/NcEmojiPicker/NcEmojiPicker.vue [hunk 1]
```diff
@@ -109,12 +109,27 @@ This component allows the user to pick an emoji.
...
 			v-bind="$attrs"
-			@select="select" />
+			@select="select">
+			<template #searchTemplate="slotProps">
+				<NcTextField ref="search"
+					class="search"
+					:value.sync="search"
+					:label="t('Search')"
+					:label-visible="true"
+					:placeholder="slotProps.i18n.search"
```
Comment (skjnldsv) [inline src/components/NcEmojiPicker/NcEmojiPicker.vue:119]: do you actually need to use `slotProps` ?
```suggestion
					:placeholder="i18n.search"
```

Comment (Pytal) [reply src/components/NcEmojiPicker/NcEmojiPicker.vue]: Good point

Now using `i18n` directly while keeping the `slotProps.onSearch()` which is necessary for emoji search https://github.com/serebrov/emoji-mart-vue/blob/a8ac66cffd2bd08f64371add84ea71b7ef620dca/src/utils/picker.js#L97-L104

File: src/components/NcEmojiPicker/NcEmojiPicker.vue [hunk 2]
```diff
@@ -109,12 +109,27 @@ This component allows the user to pick an emoji.
...
+					class="search"
+					:value.sync="search"
+					:label="t('Search')"
+					:label-visible="true"
+					:placeholder="i18n.search"
+					trailing-button-icon="close"
+					:trailing-button-label="t('Clear search')"
+					:show-trailing-button="search !== ''"
+					@trailing-button-click="clearSearch(); slotProps.onSearch(search);"
+					@update:value="slotProps.onSearch(search)" />
```
Comment (raimund-schluessler) [inline src/components/NcEmojiPicker/NcEmojiPicker.vue:124]: Shall we maybe also show a `Magnify` icon to indicate the field is for searching?
```suggestion
					@update:value="slotProps.onSearch(search)">
					<Magnify :size="16" />
				</NcTextField>
```

**Comment by JuliaKirschenheuter** [comment]
> You mean drop these changes and keep the built-in search input?

No, of coarse not. Each improvement is very welcome! I mean that we have to do a review of emoji pickershortly before next a11y review and if emoji picker would have bigger a11y issues from there side than we could "take it away" for some version if fixes would be very height. Of coarse we can discuss any (other) conceivable solution with Michael.

**Review by JuliaKirschenheuter — APPROVED** [review]

File: src/components/NcEmojiPicker/NcEmojiPicker.vue (hunk 2 above)
Comment (Pytal) [reply src/components/NcEmojiPicker/NcEmojiPicker.vue:124]: Was considered, though the other search fields, e.g. unified search and contacts search, do not have a magnify icon so will keep it consistent here :)

**Pull request merged**